### PR TITLE
Partial Fix to the REST Client

### PR DESCRIPTION
**Pull request by Du-z**
Partial Fix to HTTP Client, still need to fix queries.

### CODE REVIEW

**Comment by Du-z** [comment]
Changes include:

- Removing `Content-Type` `charset` headers
- Adding `Content-Type` `application/json` for `GET` and `DELETE` verbs
- `Change` and `Modify` use the same `PATCH` endpoint, need more tests to see if this is correct
- Database Tests properly set IDs

File: src/Core/Database/DbRest.cs [hunk 1]
```diff
@@ -199,14 +195,12 @@ public async Task<SurrealRestResponse> Update(SurrealThing thing, HttpContent da
 
     public async Task<SurrealRestResponse> Change(SurrealThing thing, object data, CancellationToken ct = default)
     {
-        // Is this the most optimal way?
-        string sql = "UPDATE $what MERGE $data RETURN AFTER";
```
Comment (ProphetLamb) [inline src/Core/Database/DbRest.cs:203]: The `Modify` command uses JSON patch syntax https://jsonpatch.com/ in order to modify the data. `Change` replaces the values of keys, by overwriting the existing values

File: src/Core/Database/DbRest.cs [hunk 2]
```diff
@@ -164,7 +164,8 @@ public async Task<SurrealRestResponse> Query(HttpContent sql, CancellationToken
 
     public async Task<SurrealRestResponse> Select(SurrealThing thing, CancellationToken ct = default)
     {
-        var rsp = await _client.GetAsync($"key/{FormatUrl(thing)}", ct);
+        var requestMessage = ToRequestMessage(HttpMethod.Get, $"key/{FormatUrl(thing)}");
+        var rsp = await _client.SendAsync(requestMessage, ct);
```
Comment (ProphetLamb) [inline src/Core/Database/DbRest.cs:168]: We use `$"key/{FormatUrl(thing)}"` often should extract that into a separate method

**Review by ProphetLamb — CHANGES_REQUESTED** [review]

File: src/Core/Database/DbRest.cs (hunk 1 above)
Comment (Du-z) [reply src/Core/Database/DbRest.cs:203]: Ahh! understood, I have reverted this change.

File: src/Core/Database/DbRest.cs (hunk 2 above)
Comment (Du-z) [reply src/Core/Database/DbRest.cs:168]: no prob, that's done

Comment (ProphetLamb) [reply src/Core/Database/DbRest.cs:168]: Looking good here. Thank you :)

**Review by ProphetLamb — APPROVED** [review]

**Pull request merged**